packages/web/src/pages/search-page-v2/RecentSearches.tsx

```diff
@@ -0,0 +1,318 @@
+
+  if (status === Status.LOADING) return <RecentSearchSkeleton />
+
+  if (!track) return null
+  const { permalink, title, user } = track
+
+  if (!user) return null
+
+  return (
+    <RecentSearch searchItem={searchItem} title={title} linkTo={permalink}>
+      <Artwork src={image} w='40px' borderRadius='xs' />
+      <Flex direction='column' css={{ alignItems: 'flex-start' }}>
+        <Text variant='body' size='s'>
+          {title}
+        </Text>
+        <Flex alignItems='baseline'>
+          <Text variant='body' size='xs' color='subdued'>
+            {messages.track}
+            {' |'}
+            &nbsp;
+          </Text>
+          <UserNameAndBadges
+            renderName={(name) => (
+              <Text variant='body' size='xs' color='subdued'>
+                {name}
+              </Text>
+            )}
+            userId={user.user_id}
+          />
+        </Flex>
+      </Flex>
+    </RecentSearch>
+  )
+}
+
+const RecentSearchCollection = (props: { searchItem: SearchItem }) => {
+  const { searchItem } = props
+  const { id } = searchItem
+  const { data: playlist, status } = useGetPlaylistById({
+    playlistId: id
+  })
+
+  const image = useCollectionCoverArt2(
+    playlist?.playlist_id,
+    SquareSizes.SIZE_150_BY_150
+  )
+
+  if (status === Status.LOADING) return <RecentSearchSkeleton />
+
+  if (!playlist) return null
+  const { is_album, playlist_name, permalink, user } = playlist
+
+  if (!user) return null
+
+  return (
+    <RecentSearch
+      searchItem={searchItem}
+      title={playlist_name}
+      linkTo={permalink}
+    >
+      <Artwork src={image} w='40px' borderRadius='xs' />
+      <Flex direction='column' css={{ alignItems: 'flex-start' }}>
+        <Text variant='body' size='s'>
+          {playlist_name}
+        </Text>
+        <Flex alignItems='baseline'>
+          <Text variant='body' size='xs' color='subdued'>
+            {is_album ? messages.album : messages.playlist}
+            {' |'}
+            &nbsp;
+          </Text>
+          <UserNameAndBadges
+            renderName={(name) => (
+              <Text variant='body' size='xs' color='subdued'>
+                {name}
+              </Text>
+            )}
+            userId={user.user_id}
+          />
+        </Flex>
+      </Flex>
+    </RecentSearch>
+  )
+}
+
+type RecentSearchUserProps = {
+  searchItem: SearchItem
+}
+
+const RecentSearchUser = (props: RecentSearchUserProps) => {
+  const { searchItem } = props
+  const { id } = searchItem
+  const { data: user, status } = useGetUserById({ id })
+
+  const image = useProfilePicture(id, SquareSizes.SIZE_150_BY_150)
+
+  if (status === Status.LOADING) return <RecentSearchSkeleton />
+
+  if (!user) return null
+  const { handle, name } = user
+
+  return (
+    <RecentSearch
+      searchItem={searchItem}
+      title={name}
+      linkTo={profilePage(handle)}
+    >
+      <Avatar src={image} w='40px' borderWidth='thin' />
```

Comment:
use web's avatar, you just need to pass userId. Id like us to add the same for CollectionArtwork too